File: backend/components/metiers/threed/plugins/core/advanced_plugin.js
// advanced_plugin.js – Plugin avancé JS (exemple clé en main)
class AdvancedPlugin {
  constructor() {
    this.activated = false;
    this.auditTrail = [];
  }
  activate(ctx) {
    this.activated = true;
    this.auditTrail.push({ event: 'activate', ctx });
  }
  run(data) {
    if (!this.activated) throw new Error('Plugin non activé');
    this.auditTrail.push({ event: 'run', data });
    return { ...data, plugin: 'advanced', status: 'ok' };
  }
  getAuditTrail() {
    return this.auditTrail;
  }
}
module.exports = { AdvancedPlugin };

File: backend/components/metiers/threed/plugins/core/advanced_plugin.py
# advanced_plugin.py – Plugin avancé Python (exemple clé en main)
class AdvancedPlugin:
    def __init__(self):
        self.activated = False
        self.audit_trail = []
    def activate(self, ctx):
        self.activated = True
        self.audit_trail.append({'event': 'activate', 'ctx': ctx})
    def run(self, data):
        if not self.activated:
            raise Exception('Plugin non activé')
        self.audit_trail.append({'event': 'run', 'data': data})
        result = data.copy() if isinstance(data, dict) else {'value': data}
        result['plugin'] = 'advanced'
        result['status'] = 'ok'
        return result
    def get_audit_trail(self):
        return self.audit_trail
